test/unit/host_test.rb

```diff
@@ -1907,7 +1907,32 @@ def to_managed!
     assert host.errors[:provision_method].include?("can't be updated after host is provisioned")
   end
 
-  test "#image_build? must be true when provision_method is image" do
+  test "#provision_methods must include build and image by default" do
+    host = FactoryGirl.create(:host, :managed)
+    assert host.provision_methods.key?('build')
+    assert host.provision_methods.key?('image')
+  end
+
+  test "#registered_provision_methods must return empty hash if no plugins are initialized" do
+    host = FactoryGirl.create(:host, :managed)
+    assert_equal host.registered_provision_methods, {}
+  end
+
+  test 'validation of a host should work with a newly registered provision method' do
+    Host::Managed.any_instance.stubs(:capabilities).returns([:build, :awesome])
+    Foreman::Plugin.register :awesome_provision do
+      name 'Awesome provision'
+      provision_method 'awesome', 'Awesomeness Based'
+    end
+    host = FactoryGirl.create(:host,
```

Comment:
ditto, use .build and then you can also move the stub to `host.expects(:capabilities)..` (I think you're using any_instance because creating the object will validate it in the process)